src/OpenTelemetry/Metrics/MetricPoint.cs

```diff
@@ -288,6 +305,60 @@ internal void TakeSnapShot(bool outputDelta)
                         break;
                     }
             }
+
+            this.MetricPointStatus = MetricPointStatus.NoPendingCollect;
+        }
+
+        internal void MarkStale()
+        {
+            if (this.MetricPointStatus == MetricPointStatus.NoPendingCollect)
+            {
+                lock (this.lockObject)
+                {
+                    if (this.MetricPointStatus == MetricPointStatus.NoPendingCollect)
+                    {
+                        this.MetricPointStatus = MetricPointStatus.CandidateForRemoval;
+                    }
+                }
+            }
+        }
+
+        internal bool TryFree()
+        {
+            if (this.MetricPointStatus == MetricPointStatus.CandidateForRemoval)
+            {
+                lock (this.lockObject)
+                {
+                    if (this.MetricPointStatus == MetricPointStatus.CandidateForRemoval)
+                    {
+                        this.MetricPointStatus = MetricPointStatus.Unused;
+                        return true;
+                    }
+                }
+            }
+
+            return false;
+        }
+
+        private bool TrySetCollectPending()
+        {
+            if (this.MetricPointStatus == MetricPointStatus.CandidateForRemoval || this.MetricPointStatus == MetricPointStatus.Unused)
```

Comment:
we need to take lock for NoPendingCollect as well. Else MarkStale might just override this to CandidateForRemoval

Reply:
Yea, you're right. I was hoping there'd be a way to keep the update path lock free.